core/src/main/scala/chisel3/internal/firrtl/Converter.scala

```diff
@@ -281,6 +281,12 @@ private[chisel3] object Converter {
           convert(elseRegion, ctx, typeAliases)
         )
       )
+    case Region(info, region) =>
+      if (region.isEmpty) {
+        None
+      } else {
+        Some(fir.Block(convert(region, ctx, typeAliases)))
+      }
```

Comment:
```suggestion
      Option.when(region.nonEmpty) {
        fir.Block(convert(region, ctx, typeAliases))
      }
```
YMMV but I like `Option.when`

Reply:
This cleans up the code as written. There is actually a slightly different fix that is necessary here as this needs to return `None` only if it isn't handled. I.e., this should just be a `Some`. Will push a fix and land.

Reply:
Fixed in: https://github.com/chipsalliance/chisel/pull/4188/commits/3738c1fa0102240f370781dab0c595fe15e66ff0